src/backend/distributed/executor/citus_custom_scan.c

```diff
@@ -129,11 +133,17 @@ CitusBeginScan(CustomScanState *node, EState *estate, int eflags)
 	ExecInitResultSlot(&scanState->customScanState.ss.ps, &TTSOpsMinimalTuple);
 #endif
 
-	distributedPlan = scanState->distributedPlan;
+	DistributedPlan *distributedPlan = scanState->distributedPlan;
+	Job *workerJob = distributedPlan->workerJob;
+	if (workerJob &&
+		(workerJob->requiresMasterEvaluation || workerJob->deferredPruning))
```

Comment:
I actually find this easier to read in the sense that we get into string generation only when `workerJob->requiresMasterEvaluation || workerJob->deferredPruning`

The check for `workerJob ` being not null, that is unfortunate and very nice, I agree. That's because for INSERT .. SELECT via coordinator, Citus doesn't create a Job. 

Are you OK if I keep it like this?

Reply:
I think it's okay to keep as it makes the intention clearer here indeed. But in that case there should at least be an `Assert` in the other function to ensure this invariant is not broken in the future.